common/vendor/MarkdownEditor/angular-markdown-editor.js

```diff
@@ -0,0 +1,200 @@
+                  showButtons: options.showButtons || null,
+                  additionalButtons: options.additionalButtons || (options.addExtraButtons ? addNewButtons() : []),
+
+                  //-- Events/Hooks --
+                  // each of them are defined as callback available in the directive
+                  // example: <textarea markdown-editor="{'iconlibrary': 'fa'}" on-fullscreen-exit="vm.exitFullScreenCallback()"></textarea>
+                  //  NOTE: If you want this one to work, you will have to manually download the JS file, not sure why but they haven't released any versions in a while
+                  //       https://github.com/toopay/bootstrap-markdown/tree/master/js
+                  onPreview: function (e) {
+
+                      var previewContent = ERMrest.renderMarkdown(e.getContent());
+                      runScopeFunction(scope, attrs.onPreview, e);
+                      return previewContent;
+                   },
+                  onPreviewEnd: function (e) { runScopeFunction(scope, attrs.onPreviewEnd, e); },
+                  onSave: function (e) { runScopeFunction(scope, attrs.onSave, e); },
+                  onBlur: function (e) { runScopeFunction(scope, attrs.onBlur, e); },
+                  onFocus: function (e) { runScopeFunction(scope, attrs.onFocus, e); },
+                  onFullscreen: function (e) { runScopeFunction(scope, attrs.onFullscreen, e); },
+                  onSelect: function (e) { runScopeFunction(scope, attrs.onSelect, e); },
+                  onFullscreenExit: function (e) { runScopeFunction(scope, attrs.onFullscreenExit, e); },
+                  onChange: function(e) {
+                    // When a change occurs, we need to update scope in case the user clicked one of the plugin buttons
+                    // (which isn't the same as a keydown event that angular would listen for).
+                    ngModel.$setViewValue(e.getContent());
+
+                    runScopeFunction(scope, attrs.onChange, e);
+                  },
+                  onShow: function (e) {
+                    // keep the Markdown Object in $rootScope so that it's available also from anywhere (like in the parent controller)
+                    // we will keep this in an object under the ngModel name so that it also works having multiple editor in same controller
+                    $rootScope.markdownEditorObjects = $rootScope.markdownEditorObjects || {};
+                    $rootScope.markdownEditorObjects[ngModel.$name] = e;
+                    if (!!attrs.onShow) {
+                      runScopeFunction(scope, attrs.onShow, e);
+                    }
+                  }
+                });
+            }
+        }
+    };
+}]);
+
+/**
+ * Add new extra buttons: Strikethrough & Table
+ * @return mixed additionButtons
+ * Few buttons are commented to be added in future versions.
+ */
+
+function addNewButtons() {
+  return [[
+        // {

```

Comment:
These are buttons which are not included right now but might be useful to include in future. I left it commented for future enhancements.